src/txn/txn.c

```diff
@@ -196,13 +196,13 @@ __wt_txn_update_oldest(WT_SESSION_IMPL *session)
 	oldest_session = NULL;
 	prev_oldest_id = txn_global->oldest_id;
 
-	/* For pure read-only workloads, avoid scanning. */
-	if (prev_oldest_id == current_id) {
-		/* Check that the oldest ID has not moved in the meantime. */
-		if (prev_oldest_id == txn_global->oldest_id &&
-		    txn_global->scan_count == 0)
```

Comment:
Another way of saying that is that there is no point doing anything to update the oldest ID if we see no active transactions, regardless of whether another thread is scanning the table.